dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hive/HiveMetadataConnector.java

```diff
@@ -342,6 +342,39 @@ protected String toCallDescription() {
     }
   }
 
+  private static class FunctionsJsonlTask extends AbstractHiveMetadataTask {
+
+    private FunctionsJsonlTask() {
+      super("functions.jsonl", unused -> true);
+    }
+
+    @Override
+    protected void run(@Nonnull Writer writer, @Nonnull ThriftClientHandle thriftClientHandle)
+        throws Exception {
+      try (HiveMetastoreThriftClient client =
+              thriftClientHandle.newClient("functions-jsonl-task-client");
+          RecordProgressMonitor monitor =
+              new RecordProgressMonitor("Writing functions to " + getTargetPath())) {
+        ImmutableList<String> functions = client.getFunctionsAsJsonl();
```

Comment:
The serializer call should be here, and we should serialize and write one at a time, rather than building an entire list of strings. That will halve our memory usage.

Reply:
I've moved the serialization to the latest moment possible just before writing a single entry to the output `jsonl` file.